pytest_allure_dsl.py

```diff
@@ -275,6 +291,9 @@ def step(self, key, **kwargs):
         else:
             step_name = step
 
+        if key not in self._steps_was_used:
```

Comment:
You will not need this if statement if  _steps_was_used is a set.

Reply:
Yes, but set requires more memory :)
```
In [11]: set().__sizeof__()                                                                                                                                                                                                                  
Out[11]: 200

In [12]: list().__sizeof__()                                                                                                                                                                                                                 
Out[12]: 40
```